be/src/storage/lake/types_fwd.h

```diff
@@ -48,6 +49,10 @@ using segment_rowid_t = uint32_t;
 using DeletesMap = std::unordered_map<uint32_t, std::vector<segment_rowid_t>>;
 using DelVectorPtr = std::shared_ptr<DelVector>;
 
+using KeyIndex = size_t;
+using KeyIndexSet = std::vector<KeyIndex>;
```

Comment:
this has be changed between set and vector multiple times, maybe we just name it to KeyIndexContainer or KeyIndexCollection?